### PR TITLE
fix RobotState::computeCartesianPath

### PR DESCRIPTION
**Pull request by rhaschke**
- consider both, translational and rotational distance to compute number of interpolation of steps
- ensure that quaternions are on same half sphere before doing quaternion slerp

### CODE REVIEW

**Comment by davetcoleman** [comment]
+1 

@mlautman please review as this has affected you

**Comment by rhaschke** [comment]
@mlautman: Ping.

File: moveit_core/robot_state/src/robot_state.cpp [hunk 1]
```diff
@@ -1924,16 +1924,27 @@ double moveit::core::RobotState::computeCartesianPath(const JointModelGroup* gro
...
+  {
+    target_quaternion.x() *= -1.0;
+    target_quaternion.y() *= -1.0;
+    target_quaternion.z() *= -1.0;
+    target_quaternion.w() *= -1.0;
+  }
+
   // decide how many steps we will need for this trajectory
-  double distance = (rotated_target.translation() - start_pose.translation()).norm();
+  distance = std::max((rotated_target.translation() - start_pose.translation()).norm(),
```
Comment (mlautman) [inline moveit_core/robot_state/src/robot_state.cpp]: I really don't like that this is comparing distance in meters to distance radians. Wouldn't it be better to have an optional max_orientation_step parameter in addition to a max_cartesian_step?

Comment (rhaschke) [reply moveit_core/robot_state/src/robot_state.cpp]: I agree. However, usually we want to stick with the existing API. In the intended rewrite of this whole API, we should consider two parameters...

Comment (mlautman) [reply moveit_core/robot_state/src/robot_state.cpp]: Perhaps we should add a `TODO` for this.

Comment (mlautman) [reply moveit_core/robot_state/src/robot_state.cpp]: I'm dealing with the same thing for #798  To deal with this I added additional function headers for the new api and used a struct to minimize code duplication

Comment (rhaschke) [reply moveit_core/robot_state/src/robot_state.cpp]: I added the TODO  note.

**Comment by davetcoleman** [comment]
@rhaschke I believe you merged this with only one approval (mine) - we have a 2 reviewer policy

**Comment by rhaschke** [comment]
@davetcoleman If I understand our policy correctly, one reviewer is sufficient if the PR originates from a maintainer. As this was a rather small change, I merged after your approval. I applied this strategy to several other small PRs as well.

**Comment by v4hn** [comment]
> one reviewer is sufficient if the PR originates from a maintainer.

yes.

**Comment by davetcoleman** [comment]
I had forgotten that rule, ok. However my approval was only partial, hence my "+1" and not the approval button. I had skimmed it but then asked @mlautman to review since he's more involved with this than I. However I did not convey this clearly at all, my bad. I'll avoid using "+1" in the future unless I really mean it.

**Pull request merged**